python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -517,3 +561,52 @@ def _private_display(self, reg_slice, this_format, default_width, markup):
             ds += markup(this_format % tuple(vals), colour)
 
         return ds
+
+    @staticmethod
+    def jobHasChildrenTest(jobDirectory, datafileName):
+        """ Return True/False if given (job?) object has children associated with it
+        Args:
+            jobDirectory (str): name of folder to be examined
+            datafileName (str): name of the files containing the xml, i.e. 'data' by convention
+        """
+
+        if not path.isdir(jobDirectory):
+            return False
+        else:
+            return bool(SubJobXMLList.countSubJobDirs(jobDirectory, datafileName, True))
+
+    @staticmethod
+    def countSubJobDirs(jobDirectory, datafileName, checkDataFiles):
+        """ I'm a function which returns a number, my number corresponds to the amount of sequentially listed numerically named folders exiting within 'jobDirectory'
+            This (optionally) checks for the existance of all of the XML files. This is useful during a call to 'jobHasChildrenTest' but not when calling __len__ repeatedly
+        Args:
+            jobDirectory (str): name of folder to be examined
+            datafileName (str): name of the files containing the xml, i.e. 'data' by convention
+            checkDataFiles (bool): if True check for the existance of all of the data files and check this against the numerically named folders
+        """
+
+        jobDirectoryList = listdir(jobDirectory)
+
+        subjob_count=0
+        for dir_entry in jobDirectoryList:
+            if dir_entry.isdigit():
+                sj_dir = path.join(jobDirectory, dir_entry)
+                if path.isdir(sj_dir):
+                    if checkDataFiles:
+                        data_file_path = path.join(sj_dir, datafileName)
+                        if path.isfile(data_file_path):
+                            subjob_count+=1
+                        else:
+                            if path.isfile(data_file_path+'~'):
+                                logger.warning("Reverting to backup due to missing XML: %s" % data_file_path)
+                                subjob_count+=1
+                    else:
+                        subjob_count+=1
+
+        logger.debug("count: %s len: %s" % (subjob_count, len([_folder for _folder in jobDirectoryList if _folder.isdigit()])))
+
+        if subjob_count == len([_folder for _folder in jobDirectoryList if _folder.isdigit()]):
+            return subjob_count
+        else:
+            raise GangaException("Missing subjobs data file in %s" % jobDirectory)
```

Comment:
Should this be `GangaException`? I forget the rule exactly but I thought that `GangaException` was reserved for things that the user directly caused (like missing arguments to functions etc.) rather than behind-the-scenes things.